### DISCUSSION TITLE
what is future of connectionless/implicit execution?

### DISCUSSION THREAD

**Discussion by gdub** [python-gino/gino#757] [Q&A] [answered]
Reading through Gino's [Engine and Connection documentation](https://python-gino.org/docs/en/master/explanation/engine.html), there is link to SQLAlchemy's [implicit execution documentation](https://docs.sqlalchemy.org/en/14/core/connections.html#connectionless-execution-implicit-execution) which currently displays a deprecation warning message:
> Deprecated since version 2.0: The features of “connectionless” and “implicit” execution in SQLAlchemy are deprecated and will be removed in version 2.0. See “Implicit” and “Connectionless” execution, “bound metadata” removed for background.

Given the warning, I was wondering what the thought was for Gino's own implicit execution features. Will Gino's implicit execution style also be dropped?

**Comment by fantix** [comment, marked as answer]
> Will Gino's implicit execution style also be dropped?

Likely not. The [contextual connection model](https://python-gino.org/docs/en/master/explanation/engine.html#managing-connections) is one of GINO's core features (inherited from pallets/flask-sqlalchemy#447), and it's major use case is implicit execution. We will be able to do that without upstream support.

But I'm also open to hear from the community if the reusing/reusable/lazy connections are useful, if this trade of explicitness for conveniece is worth it, as the explicit programming style in async is more popular now.
